util/util.go

```diff
@@ -659,3 +659,22 @@ func ManagerDelegatingClientFunc(cache cache.Cache, config *rest.Config, options
 		StatusClient: c,
 	}, nil
 }
+
+// LowestNonZeroResult compares two reconciliation results
+// and returns the one with lowest requeue time.
+func LowestNonZeroResult(i, j ctrl.Result) ctrl.Result {
```

Comment:
What about 
```suggestion
func LowestRequeueTime(i, j ctrl.Result) ctrl.Result {
```

Reply:
Ok for keeping the name as it is.
Waiting for @vincepri to comment on 

> When Requeue has not been set and comparing to different RequeueAfter values, I think we might actually want to choose the longer of the two intervals

Reply:
Would it be ok to stick with the lowest requeue time? Preferring more requeues has been a good trade-off so far. If we have an operation X that requires an immediate requeue, but there is operation Y that requires a longer requeue, it's probably ok to requeue immediately first and then try to reconcile the rest of the operations. More often than not, our operations are going to be stacked in a chain.

Choosing the lowest requeue-time helps us not diverge within this PR too much from the earlier implementation, I'm happy to revisit this in v1alpha4 for whatever it makes sense, but given this is patch for v0.3.x, I'd vote to keep the behavior as close as possible for now

Reply:
This seems reasonable to me
/lgtm

leaving the last world to @detiber

Reply:
sgtm